### ISSUE TITLE
Tracking progress of .wasm fetch

### ISSUE THREAD

**Issue by connorjclark** [WebAssembly/esm-integration#62]
The current `fetch` approach of loading a wasm module allows you to track the progress of the (often large) network resource, and indicate its progress to the user (you just wrap `fetch` with something like [this](https://gist.github.com/connorjclark/6afb9fb588331a23a2d8fa57cfefe8f5)).

The whole point of this proposal is to make the wasm exports available "statically", so the entire loading phase is made opaque. But I wonder if a dynamic import could somehow provide hooks to track progress?

```js
const wasm = import('some.wasm');

wasm.onProgress(...)

const {exportedFn} = await wasm;
```

I understand this is not possible with dynamic imports today, so this is probably the wrong place to talk about this idea... In any case, it'd be good to mention "indicate progress" use case as a non-goal, as it's the first thing I thought of (and is a reason I wouldn't make use of this if the binary is large enough).

**Comment by annevk** [comment]
This seems like a good thing to raise with TC39: https://github.com/tc39/proposals.

**Comment by guybedford** [comment]
Agreed, this seems like a more general TC39 feature to raise there. It's definitely an interesting suggestion!

Closing here as part of issue queue cleanup now, but please do consider bringing it up there further.

**Comment by michaelficarra** [comment]
@connorjclark @annevk The appropriate place to raise this would be [TC39 Discourse](https://es.discourse.group/), not the proposals repo. See [the contributing docs](https://github.com/tc39/ecma262?tab=contributing-ov-file#new-feature-proposals).
